vault/diagnose/storage_checks.go

```diff
@@ -0,0 +1,82 @@
+package diagnose
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/physical"
+)
+
+var success string = "success"
+var secretKey string = "diagnose"
+var secretVal string = "diagnoseSecret"
+
+const timeOutErr string = "storage call timed out after 20 seconds: "
+const wrongRWValsPrefix string = "Storage get and put gave wrong values: "
+
+// StorageEndToEndLatencyCheck calls Write, Read, and Delete on a secret in the root
+// directory of the backend.
+// Note: Just checking read, write, and delete for root. It's a very basic check,
+// but I don't think we can necessarily do any more than that. We could check list,
+// but I don't think List is ever going to break in isolation.
+func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error {
+
+	c2 := make(chan string, 1)
```

Comment:
i don't think this channel needs to be buffered with the current code. you could also do a channel of errors (with nil check instead of success stirng) which would simplify the code a bit

same with `c5`

Reply:
True, will do!